clippy_lints/src/utils/conf.rs

```diff
@@ -514,6 +515,11 @@ define_Conf! {
     ///
     /// The byte size a `T` in `Box<T>` can have, below which it triggers the `clippy::unnecessary_box` lint
     (unnecessary_box_size: u64 = 128),
+    /// Lint: SINGLE_LETTER_IDENTS.
+    ///
+    /// Allowed single letter idents.
+    (allowed_idents: rustc_data_structures::fx::FxHashSet<char>
```

Comment:
Maybe another config option for a minimal length identifier? Instead of having another lint, I think having compact here is the best option.

If you're going to implement it, lints when ident's length i 1 should be different from messages when ident's length < config. So it's clear that the configuration value is what caused that message.